### ISSUE TITLE
Auto-generate docs of required privileges per statement

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/cockroach#25294]
The docs for each SQL statement list the privileges required. The source file for each statement should always, or almost always, include a comment about required privileges, for example: https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/show_jobs.go#L24. We should consider auto-generating files to include into the SQL docs based on these comments.

cc @knz

**Comment by asubiotto** [comment]
@jseldess removing from SQL Execution board since I think you'll have someone working on this soon.

**Comment by jseldess** [comment]
This is still relevant.

**Comment by fqazi** [comment]
@rmloveland Is this still relevant?

**Comment by rmloveland** [comment]
> Is this still relevant?

Yes -  If there's a way to autogenerate this info that would be great!

Right now we write out the required privileges for each statement by hand (such as in [`ALTER TABLE ... ADD COLUMN`](https://www.cockroachlabs.com/docs/dev/alter-table#add-column) - see the **Required Privileges** section) whenever we add/update docs which is pretty error prone. The workflow is basically "ask the engineer adding/changing it"

Once the information was exposed in some way in the `cockroachdb/cockroach` repo we would then figure out how to consume it from the docs site.  In some cases we just do an "include" from a Github URL right into our docs, such as for generating the [list of cluster settings](https://www.cockroachlabs.com/docs/stable/cluster-settings) (which can change frequently).  That page is pulling the list from https://github.com/cockroachdb/cockroach/blob/master/docs/generated/settings/settings.html
